### ISSUE TITLE
Device that gets connected to, is unable to check for available messages

### ISSUE THREAD

**Issue by jpinto321** [kenjdavidson/react-native-bluetooth-classic#243]
**Mobile Device Environment**
Provide a list of operating systems on which this issue is relevant.
- Device: Two Android 12 phones
- OS: Android 12

**Application Environment**
Provide information about your development environment:
- React Native version: 0.68.5
- RN Bluetooth Classic version: 1.60.0-rc.25

**Describe the bug**
**device1**: accept connection
**device2**: start discovery
**device2**: finds device1
**device2**: connects to device1
**device1/device2**: accept pairing on both devices
**device2**: check for bonded devices, which returns device1. Then check for available messages from device1 and get no available messages (this is working correctly)
**device1**: check for bonded devices, which returns device2. Then check for available messages and get an error saying device1 is not connected to device2

**Expected behavior**
I expect to be able to check for available messages on device1, because it says they are bonded.

Am I missing something here? It seems like things are only working correctly on the device that runs the "connectToDevice", while the device that gets connected to (device1), doesn't work correctly?

Also, why is it that I can only find device1 when it's on the Bluetooth settings screen, but not on the app itself. Is this normal behavior? This means that users have to move away from the app, go to the bluetooth settings, to be discoverable, and once they get connected to, they go back to the app. This seems like too much work?

**Comment by kenjdavidson** [comment]
I suggest you go back to the documentation.

tldr;
- Bonded = paired
- Connected = open RFCOMM socket

There will never be any communication without being connected.  This seems to be working as designed, I suggest you provide specific code for review if you want more assistance.

With regards to your question on settings screen vs app screen, you need to provide code and/or screenshots for me to give any realistic answers.

The library was designed without phone/phone communication in mind, it was just a happy side effect that it allowed this to work.  The negative is that you still need to go into bluetooth settings to enable broadcast.  The good thing is that this only happens once, once you're paired you're paired going forward.

I believe there is an issue open, and I'll accept a PR, with regards to adding the functionality for turning on broadcasting/discoverability in the library.

**Comment by jpinto321** [comment]
Does that mean device1 will also need to do "connectToDevice(device2)"? Even though device2 already did it.

**Comment by jpinto321** [comment]
When I say device2 connects to device1, I'm doing this on device2: **peer.connect()**, where the peer is device1.
That's when I get the notification, on both phones, to pair.
If connect is just for pairing, which method is for actually connecting?

**Comment by kenjdavidson** [comment]
So are the two phones paired or not paired?  You have two options for pairing, personally I would do it in the Android Settings as I don't particularly trust the libraries pairing functionality between phones, I've only ever used it for actual devices.

So assuming the devices are paired, they show up in each others bluetooth list.

1. On both devices doing a `BluetoothClassic.getBondedDevices` gets you a list including the other device??
2. Once you have both devices you would:

On device 1 do `BluetoothClassic.accept()`
On device 2 do `device1.connect()` or `BluetoothClassic.connectToDevice(device1.address)`

both of these should return back on their respective devices.  This should NOT pop up a message asking you to pair, if it does, then you aren't paired, and you can therefore not actually connect.

Once you have the connection you should be able to read and write as you wish.

**Comment by jpinto321** [comment]
Ah!

If I pair them manually, by going to the bluetooth settings, on both devices, then do device1.connect() from device2, and then check for available messages on both devices, I no longer get the "not connected to device2" error.

When the devices are not paired, **peer.connect()** will trigger the pairing, but not the connection (socket), while if I do connect when they are already paired (from the Bluetooth settings screen), then it will actually connect the devices. 

Thanks for your help!

**Comment by kenjdavidson** [comment]
Ya, this is a finicky thing in Android where it opens the pairing dialog, but it's not handling the connect correctly (it should fail the connect) but this is deeper in the BluetoothAdapter than I've had time to debug.

Glad it's up and running.
